From 9875c9c891322dca0764d446b2f49adcb7920c78 Mon Sep 17 00:00:00 2001
From: "Torne (Richard Coles)" <torne@google.com>
Date: Wed, 16 Oct 2013 11:58:08 +0100
Subject: [PATCH] Support a prebuilt WebView.

If the product sets PRODUCT_PREBUILT_WEBVIEWCHROMIUM, do not attempt to
build most of the WebView from source; build only
libwebviewchromium_plat_support.

Bug: 10824652
Change-Id: I236fd98916f2a20006c946c0a0c8b317e77dafd1
---
 chromium/Android.mk | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/chromium/Android.mk b/chromium/Android.mk
index d715221..bdc5b28 100644
--- a/chromium/Android.mk
+++ b/chromium/Android.mk
@@ -19,6 +19,9 @@
 LOCAL_PATH := $(call my-dir)
 CHROMIUM_PATH := external/chromium_org
 
+# Don't include most modules if the product is using a prebuilt webviewchromium.
+ifneq ($(PRODUCT_PREBUILT_WEBVIEWCHROMIUM),yes)
+
 # Java glue layer JAR, calls directly into the chromium AwContents Java API.
 include $(CLEAR_VARS)
 
@@ -114,10 +117,11 @@ $(jar_check_ok): $(full_classes_jarjar_jar) $(LOCAL_PATH)/tools/jar_check.py $(L
 $(LOCAL_BUILT_MODULE): $(jar_check_ok)
 endif
 
-
+endif # PRODUCT_PREBUILT_WEBVIEWCHROMIUM
 
 # Native support library (libwebviewchromium_plat_support.so) - does NOT link
-# any native chromium code.
+# any native chromium code. This is built from source even if the product has
+# a prebuilt webviewchromium to ensure ABI compatibility.
 include $(CLEAR_VARS)
 
 LOCAL_MODULE:= libwebviewchromium_plat_support
-- 
2.5.0

